terraform/modules/task-definition/main.tf

```diff
@@ -8,6 +8,8 @@ terraform {
 }
 
 locals {
+  virtual_node_name = var.mesh_subdomain != "" ? var.mesh_subdomain : var.service_name
```

Comment:
optional: consider writing this as `coalesce(var.mesh_subdomain, var.service_name)`

Reply:
Thanks, that's way nicer 👍